src/MeshKernel/Mesh.cpp

```diff
@@ -548,72 +548,80 @@ bool Mesh::IsFaceOnBoundary(size_t face) const
     return isFaceOnBoundary;
 }
 
-void Mesh::SortEdgesInCounterClockWiseOrder(size_t node)
+void Mesh::SortEdgesInCounterClockWiseOrder(size_t startNode, size_t endNode)
 {
-    if (!m_nodes[node].IsValid())
-    {
-        throw std::invalid_argument("Mesh::SortEdgesInCounterClockWiseOrder: Invalid nodes.");
-    }
 
-    double phi0 = 0.0;
-    double phi;
     std::vector<double> edgeAngles(maximumNumberOfEdgesPerNode);
-    std::fill(edgeAngles.begin(), edgeAngles.end(), 0.0);
-    for (auto edgeIndex = 0; edgeIndex < m_nodesNumEdges[node]; edgeIndex++)
+    std::vector<std::size_t> indices;
+    std::vector<size_t> edgeNodeCopy;
+    for (auto n = startNode; n <= endNode; n++)
     {
-
-        auto firstNode = m_edges[m_nodesEdges[node][edgeIndex]].first;
-        auto secondNode = m_edges[m_nodesEdges[node][edgeIndex]].second;
-        if (firstNode == sizetMissingValue || secondNode == sizetMissingValue)
+        if (!m_nodes[n].IsValid())
         {
-            continue;
+            throw std::invalid_argument("Mesh::SortEdgesInCounterClockWiseOrder: Invalid nodes.");
```

Comment:
in the for-loop, some state changes occur on the this-pointer. With this exception, it may stop halfway. Does that leave the object in an invalid state? if you are going to stop program execution after this exception, that 's not a problem, but if you want to continue working with the objects, then you should avoid an invalid state, and perform the checks before starting.